parser.rb

```diff
@@ -1,24 +1,40 @@
+require 'json'
+
 class ParserGamesLogs
   def initialize(filename_with_path)
     @filename_with_path = filename_with_path
     @mode = 'r'
-    file_exists?()
+    file_exists()
   end
   
 
   def read_first_line()
-    if file_exists?
+    if file_exists
       File.open(@filename_with_path, @mode) do |log_file|
         log_file.readline
       end
     end
   end
 
 
+  def parse_file()
+    count = count_lines()
+    output_hash = {@filename_with_path => { "lines" => count}}
+    return output_hash.to_json
+  end
+
+
   private
   
-  def file_exists?()
+  def file_exists()
```

Comment:
nao faz sentido isso retorna true ou quebra? pra que o if na linha12 entao?

Reply:
Não e uma boa ideia usar exceptions para controlar fluxo. E é bom separar checagem de estado de tratamento de erro.
Dá uma olhada nesse artigo aqui do blog: https://blog.codeminer42.com/and-understanding-one-of-rubys-coolest-naming-conventions-5a9300b75605/